rfi_file_monitor/operations/scicataloguer.py

```diff
@@ -0,0 +1,804 @@
+                    file.filename,
+                    parser,
+                )
+            payload.scientificMetadata = (
+                PayloadHelpers.scientific_metadata_concatenation(
+                    scientificMetadata, payload.scientificMetadataDefaults
+                )
+            )
+
+        del payload.scientificMetadataDefaults
+        return payload
+
+    def find_parser(self, filename):
+        for parser in self.parser_list:
+            if parser.supports_file(filename):
+                break
+        else:
+            parser = None
+        if not parser:
+            raise ParserNotFound("parser not found")
+        return parser
+
+    # Inserts a dataset into Scicat
+    def insert_payload(self, payload, scicat_session):
+        try:
+            r = scicat_session.upload_dataset(payload)
+            if r:
+                logger.info(f"Payload catalogued, PID: {r}")
+        except Exception as e:
+            logger.error(
+                f"Could not catalogue payload in scicat: {e.statusCode} -> {e.message}"
+            )
+            return str(e)
+
+    # Upserts dataset in Scicat
+    # This won't work with upserting until features added into PySciCat
+    def upsert_payload(self, payload, scicat_session):
+        # Ensure that raw/derived datasets don't overwrite each other
+        query_results = scicat_session.get_datasets(
+            {"datasetName": payload.datasetName, "type": payload.type}
+        )
+        if query_results:
+            if query_results[0]["datasetName"] == payload.datasetName:
+                logger.info("pretending to upsert payload")
+                # try:
+                # if payload.type == "raw":
+                # r = scicat_session.upsert_raw_dataset(payload, {"datasetName": payload.datasetName, "type": payload.type})
+                # else:
+                # r = scicat_session.upsert_derived_dataset(payload, {"datasetName": payload.datasetName, "type": payload.type})
+                # if r:
+                #    logger.info(f"Payload upserted, PID: {r}")
+                # except Exception as e:
+                #   logger.error(f"Could not catalogue payload in scicat: {e}")
+                #   return str(e)
+            else:
+                self.insert_payload(payload, scicat_session)
+        else:
+            self.insert_payload(payload, scicat_session)
+
+
+# Base Payload Model inherits from Dataset Model
+class Payload(Dataset):
+    datasetlifecycle = {"retrievable": True}
+    scientificMetadataDefaults = {}
+    scientificMetadataDefaults["RFI File Monitor Version"] = {
+        "type": "string",
+        "value": core_version,
+        "unit": "",
+    }
+
+
+# Extends Payload for raw data
+class RawPayload(RawDataset, Payload):
+    dataFormat: Optional[str]
+
+
+# Extends Payload for derived data
+class DerivedPayload(DerivedDataset, Payload):
+    inputDatasets: Optional[List[str]]
+    usedSoftware: Optional[List[str]]
+
```

Comment:
would be nice to have a data format in here too? Unless this is not accepted in the scicat model